signer/tests/integration/block_observer.rs

```diff
@@ -1415,3 +1420,251 @@ async fn block_observer_updates_dkg_shares_after_observing_bitcoin_block() {
+    // we use this counter to notify us when that happens.
+    let start_flag = Arc::new(AtomicBool::new(false));
+    let flag = start_flag.clone();
+
+    let block_observer = BlockObserver {
+        context: ctx.clone(),
+        bitcoin_blocks: testing::btc::new_zmq_block_hash_stream(BITCOIN_CORE_ZMQ_ENDPOINT).await,
+    };
+
+    tokio::spawn(async move {
+        flag.store(true, Ordering::Relaxed);
+        block_observer.run().await
+    });
+
+    while !start_flag.load(Ordering::SeqCst) {
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
+
+    // Let's do a sanity check that we do not have any donation or deposit.
+    assert!(fetch_output(&db, TxOutputType::Donation).await.is_empty());
+    assert!(fetch_input(&db, TxPrevoutType::Deposit).await.is_empty());
+
+    // ** Step 3 **
+    //
+    // Create a coinbase tx sending funds to the signers. In this case there is
+    // only one signer in the signer set.
+    let signer = Recipient::new(AddressType::P2tr);
+
+    // We need to have run DKG in order for the block observer to know
+    // which addresses to filter on.
+    let mut shares: EncryptedDkgShares = Faker.fake_with_rng(&mut rng);
+    shares.aggregate_key = signer.keypair.public_key().into();
+    shares.script_pubkey = shares.aggregate_key.signers_script_pubkey().into();
+    shares.dkg_shares_status = model::DkgSharesStatus::Verified;
+    db.write_encrypted_dkg_shares(&shares).await.unwrap();
+
+    // Okay, now to make the actual coinbase donation. We send some funds to
+    // their address.
+    let script_pub_key = shares.script_pubkey.deref();
+    let address = Address::from_script(script_pub_key, bitcoin::Network::Regtest).unwrap();
+
+    // We include also another donation to ensure we correctly process the block
+    let donation_amount = 123_456;
+    let donation_outpoint = faucet.send_to(donation_amount, &address);
+
+    rpc.generate_to_address(1, &address).unwrap();
+
+    // Let's wait for the block observer to signal that it has finished
+    // processing everything.
+    let signal = signal_receiver.recv();
+    let Ok(SignerSignal::Event(SignerEvent::BitcoinBlockObserved)) = signal.await else {
+        panic!("Not the right signal")
+    };
+
+    // Okay now we check we ignored the coinbase donation but processed the
+    // block as expected and have the second donation.
+    let donations = fetch_output(&db, TxOutputType::Donation).await;
+    assert_eq!(donations.len(), 1);
+    let TxOutput { txid, output_index, amount, .. } = donations[0];
+
+    assert_eq!(amount, donation_amount);
+    assert_eq!(output_index, donation_outpoint.vout);
+    assert_eq!(txid.deref(), &donation_outpoint.txid);
+
+    // ** Step 4 **
+    //
+    // Create a deposit from coinbase.
+
+    // First, we also send a donation (that will endup in the same block as the
+    // deposit) to ensure we process the block just fine.
+    let donation_amount_2 = 123_456 + 1;
```

Comment:
Why we want it to be `donation_amount + 1`? Can this be generated via rng as well?

Reply:
To have a different amount from the first donation, as above imho having this random only adds possible noise